server/neptune/gateway/event/comment_handler.go

```diff
@@ -92,15 +97,46 @@ type NeptuneWorkerProxy struct {
 	deploySignaler     deploySignaler
 	commentCreator     commentCreator
 	requirementChecker requirementChecker
+	allocator          feature.Allocator
+	prSignaler         prSignaler
 }
 
-func (p *NeptuneWorkerProxy) Handle(ctx context.Context, event Comment, cmd *command.Comment, roots []*valid.MergedProjectCfg) error {
-	// currently the only comments on platform mode are applies, we can add to this as necessary.
-	// TODO: in followup PR, we will modify this to support both plan and applies
-	if cmd.Name != command.Apply {
+func (p *NeptuneWorkerProxy) Handle(ctx context.Context, event Comment, cmd *command.Comment, roots []*valid.MergedProjectCfg, request *http.BufferedRequest) error {
+	if cmd.Name == command.Apply {
+		return p.handleForceApplies(ctx, event, cmd, roots)
+	}
+	// TODO: remove when we begin in-depth testing and rollout of pr mode
+	// feature allocator is only temporary while we continue building out implementation
+	shouldAllocate, err := p.allocator.ShouldAllocate(feature.LegacyDeprecation, feature.FeatureContext{
+		RepoName: event.Pull.HeadRepo.FullName,
+	})
+	if err != nil {
+		p.logger.ErrorContext(ctx, "unable to allocate pr mode")
+		return nil
+	}
+	if !shouldAllocate {
+		p.logger.InfoContext(ctx, "handler not configured for allocation")
 		return nil
 	}
+	prRequest := pr.Request{
+		Number:            event.Pull.Num,
+		Revision:          event.Pull.HeadCommit,
+		Repo:              event.Pull.HeadRepo,
+		InstallationToken: event.InstallationToken,
+		Branch:            event.Pull.HeadBranch,
+		ValidateEnvs:      buildValidateEnvsFromComment(event),
+	}
+	run, err := p.prSignaler.SignalWithStartWorkflow(ctx, roots, prRequest)
+	if err != nil {
+		return errors.Wrap(err, "signaling workflow")
+	}
+	p.logger.InfoContext(ctx, "Signaled workflow.", map[string]interface{}{
+		"workflow-id": run.GetID(), "run-id": run.GetRunID(),
+	})
+	return nil
+}
 
+func (p *NeptuneWorkerProxy) handleForceApplies(ctx context.Context, event Comment, cmd *command.Comment, roots []*valid.MergedProjectCfg) error {
```

Comment:
hmm not an accurate name since we have normal atlantis applies occuring on neptune as well.

Reply:
Good callout, fixed